src/main/java/com/google/api/codegen/util/ruby/RubyCommentReformatter.java

```diff
@@ -17,17 +17,44 @@
 import com.google.api.codegen.CommentPatterns;
 import com.google.api.codegen.util.CommentReformatter;
 import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
 import java.util.regex.Matcher;
 
 public class RubyCommentReformatter implements CommentReformatter {
   @Override
   public String reformat(String comment) {
-    comment = CommentPatterns.BACK_QUOTE_PATTERN.matcher(comment).replaceAll("+");
-    comment = reformatProtoMarkdownLinks(comment);
-    comment = reformatCloudMarkdownLinks(comment);
-    comment = reformatAbsoluteMarkdownLinks(comment);
-    comment = reformatHeadline(comment);
-    return comment.trim();
+    StringBuffer sb = new StringBuffer();
+    boolean inList = false;
+    for (String line : Splitter.on("\n").split(comment)) {
+      Matcher listMatcher = CommentPatterns.UNORDERED_LIST_PATTERN.matcher(line);
+      boolean matchesList = listMatcher.lookingAt();
+      if (line.isEmpty()) {
+        inList = false;
+      } else if (matchesList) {
+        inList = true;
+      }
+      if (inList) {
+        Matcher indentMatcher = CommentPatterns.INDENT_PATTERN.matcher(line);
+        if (matchesList) {
+          line = listMatcher.replaceFirst("* ");
+        } else if (indentMatcher.lookingAt()) {
```

Comment:
Ah got it. Could you comment the code here? Also, it might be good to make the test case cover more of the possible list syntax, like:

```
// +  Here is a sentence about the first element of the list that continues
//     into a second line.
// +  The second element of the list.
//
// +  Another element of the list after a blank space
// where the indentation isn't consistent
//
//    still part of the
// last list bullet.
```